File: JavaScript - C#/1- Problem Solving Patterns/Frequency Counter/countUnique.cs
using System;
using System.Collections.Generic;

class Program
{
    static int CountUniqueValues(int[] arr)
    {
        Dictionary<int, int> valueCount = new Dictionary<int, int>();

        foreach (int val in arr)
        {
            if (valueCount.ContainsKey(val))
                valueCount[val]++;
            else
                valueCount[val] = 1;
        }

        return valueCount.Count;
    }

    static void Main(string[] args)
    {
        int[] arr = { 1, 2, 3, 4, 2, 3, 4, 5, 6, 7, 8, 9, 1, 2, 3, 4, 5 };
        int uniqueCount = CountUniqueValues(arr);
        Console.WriteLine("Number of unique values: " + uniqueCount);
    }
}

File: JavaScript - C#/1- Problem Solving Patterns/Frequency Counter/countUnique.js
function countUniqueValues(arr) {
  let objTracker = {};
  let total = 0;

  for (val of arr) objTracker[val] = (objTracker[val] || 0) + 1;
  // console.log(objTracker);

  for (let key in objTracker) total++;

  return total;
}

console.log(countUniqueValues([1, 1, 1, 1, 1, 2])); //2
console.log(countUniqueValues([1, 2, 3, 4, 4, 4, 7, 7, 12, 12, 13])); //7
console.log(countUniqueValues([])); //0
console.log(countUniqueValues([-2, -1, -1, 0, 1])); //4
